### PR TITLE
Upgrade to Workbox 3.0

### PR DESCRIPTION
**Pull request by ZhangYiJiang**
Keeping pretty much the same config. Thank goodness our config is small, because the documentation and migration guide is awful.

### CODE REVIEW

File: www/package.json [hunk 1]
```diff
@@ -94,7 +94,7 @@
     "webpack": "^3.11.0",
     "webpack-dev-server": "^2.11.1",
     "webpack-merge": "4.1.2",
-    "workbox-webpack-plugin": "2.1.3"
+    "workbox-webpack-plugin": "^3.0.0"
```
Comment (taneliang) [inline www/package.json]: Probably shouldn't put the ^. IIRC I had to enable pinning so that Renovate would work properly. Not entirely sure why

**Comment by ZhangYiJiang** [comment]
Hmmm, do you have some log of the error? The new version of the plugin works for me with Chrome latest using the preview deploy

**Comment by li-kai** [comment]
Full log:
```
Uncaught (in promise) t: expire-custom-caches-only :: [null]
    at e.Plugin.b (https://storage.googleapis.com/workbox-cdn/releases/3.0.0/workbox-cache-expiration.prod.js:1:2646)
    at https://storage.googleapis.com/workbox-cdn/releases/3.0.0/workbox-cache-expiration.prod.js:1:3207
    at Generator.next (<anonymous>)
    at n (https://storage.googleapis.com/workbox-cdn/releases/3.0.0/workbox-core.prod.js:1:167)
    at https://storage.googleapis.com/workbox-cdn/releases/3.0.0/workbox-core.prod.js:1:306
    at new Promise (<anonymous>)
    at https://storage.googleapis.com/workbox-cdn/releases/3.0.0/workbox-core.prod.js:1:104
    at e.Plugin.cacheDidUpdate (https://storage.googleapis.com/workbox-cdn/releases/3.0.0/workbox-cache-expiration.prod.js:1:3268)
    at Object.<anonymous> (https://storage.googleapis.com/workbox-cdn/releases/3.0.0/workbox-core.prod.js:1:2542)
    at Generator.next (<anonymous>)
```

**Comment by ZhangYiJiang** [comment]
Ah, I figured out what this error is - 

>  You must provide a 'cacheName' property when using the expiration plugin with a runtime caching strategy.

https://github.com/GoogleChrome/workbox/blob/142da6f1c2527b4a07e9d0b838086f73b4aefdd3/packages/workbox-core/models/messages/messages.mjs

Workbox uses the same strategy React uses to strip out actual error messages in production. Unfortunately because of the way we use it, we can't use Workbox in development. I wonder if there's a way to get Workbox working with Webpack Dev Server

**Pull request merged**